plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/split/HudiSplitFactory.java

```diff
@@ -44,57 +42,74 @@ public HudiSplitFactory(
         this.hudiSplitWeightProvider = requireNonNull(hudiSplitWeightProvider, "hudiSplitWeightProvider is null");
     }
 
-    public Stream<HudiSplit> createSplits(List<HivePartitionKey> partitionKeys, HudiFileStatus fileStatus)
+    public List<HudiSplit> createSplits(List<HivePartitionKey> partitionKeys, HudiFileStatus fileStatus)
     {
-        List<FileSplit> splits;
         try {
-            splits = createSplits(fileStatus);
-        }
-        catch (IOException e) {
-            throw new TrinoException(HUDI_CANNOT_OPEN_SPLIT, e);
-        }
+            if (fileStatus.isDirectory()) {
+                throw new IOException("Not a file: " + fileStatus.path());
+            }
 
-        return splits.stream()
-                .map(fileSplit -> new HudiSplit(
-                        fileSplit.getPath().toString(),
-                        fileSplit.getStart(),
-                        fileSplit.getLength(),
-                        fileStatus.length(),
+            long length = fileStatus.length();
+
+            if (length == 0) {
+                return ImmutableList.of(new HudiSplit(
+                        fileStatus.path().toString(),
+                        0,
+                        0,
+                        length,
                         fileStatus.modificationTime(),
                         ImmutableList.of(),
                         hudiTableHandle.getRegularPredicates(),
                         partitionKeys,
-                        hudiSplitWeightProvider.calculateSplitWeight(fileSplit.getLength())));
-    }
-
-    private List<FileSplit> createSplits(HudiFileStatus fileStatus)
-            throws IOException
-    {
-        if (fileStatus.isDirectory()) {
-            throw new IOException("Not a file: " + fileStatus.path());
-        }
+                        hudiSplitWeightProvider.calculateSplitWeight(0)));
+            }
 
-        long length = fileStatus.length();
-
-        if (length == 0) {
-            return ImmutableList.of(new FileSplit(fileStatus.path(), 0, 0, new String[0]));
-        }
-
-        if (fileStatus.path() instanceof PathWithBootstrapFileStatus) {
-            return ImmutableList.of(new FileSplit(fileStatus.path(), 0, length, (String[]) null));
-        }
+            if (fileStatus.path() instanceof PathWithBootstrapFileStatus) {
```

Comment:
It looks like you can combine this with the one above, since the code and result is the same.

Reply:
done